### ISSUE TITLE
Crash while using proxy

### ISSUE THREAD

**Issue by sfred92** [siberas/watobo#4]
Hi,

Waboto 0.9.21 crashes as soon as I use it as a proxy server with Firefox. Basically, I cannot use Waboto at all. Output of Ruby is below.
My OS is Windows 10, with Ruby 2.3.1 x64 and DevKit 4.7.2 x64.

Thanks.

Fred

---

> > Request RAW (HEX):
> > !!! Error reading client request
> > undefined method `unpack' for nil:NilClass
> > C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/watobo-0.9.21/lib/watobo/sockets/client_socket.rb:107:in`request'
> > C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/watobo-0.9.21/lib/watobo/interceptor/proxy.rb:184:in `block (3 levels) in start'
> > C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/watobo-0.9.21/lib/watobo/interceptor/proxy.rb:179:in`loop'
> > C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/watobo-0.9.21/lib/watobo/interceptor/proxy.rb:179:in `block (2 levels) in start'
> > C:/Ruby23-x64/lib/ruby/gems/2.3.0/gems/watobo-0.9.21/lib/watobo/utils/secure_eval.rb:18:in`block in secure_eval': $SAFE=2 to 4 are obsolete (ArgumentError)

**Comment by siberas** [comment]
Hi Fred,

I don't have a win10 system at the moment, but I will setup up one to try to reproduce this issue.

In the meantime, have you tried the code of the git repo? this code is more up-to-date.

regards,

Andy

**Comment by siberas** [comment]
Ok, I could reproduce the crash with the watobo gem.

The current git version of watobo doesn't crash, but you will need ruby 2.2. Ruby 2.3 does not work because of nokogiri :(

The git version also uses bundler, which makes the installation of the required packages easier.

After you cloned the git repo, please follow these instructions:

```
siberas@DESKTOP-J9Q9RTS MINGW64 ~
$ cd watobo/

siberas@DESKTOP-J9Q9RTS MINGW64 ~/watobo (master)
$ ruby -v
ruby 2.2.5p319 (2016-04-26 revision 54774) [x64-mingw32]

siberas@DESKTOP-J9Q9RTS MINGW64 ~/watobo (master)
$ cd bin

siberas@DESKTOP-J9Q9RTS MINGW64 ~/watobo/bin (master)
$ ruby watobo
#############################################################

     W A T O B O - THE Web Application Toolbox
     brought to you by siberas http://www.siberas.de

#############################################################
You will need bundler to run watobo!
please run
 gem install bundler
 bundle install

siberas@DESKTOP-J9Q9RTS MINGW64 ~/watobo/bin (master)
$ gem install bundler
Successfully installed bundler-1.12.5
Parsing documentation for bundler-1.12.5
Installing ri documentation for bundler-1.12.5
Done installing documentation for bundler after 18 seconds
1 gem installed

siberas@DESKTOP-J9Q9RTS MINGW64 ~/watobo/bin (master)
$ bundle install
Fetching gem metadata from https://rubygems.org/
Fetching version metadata from https://rubygems.org/
Resolving dependencies...
Installing unf_ext 0.0.7.2
Installing fxruby 1.6.29
Installing jwt 1.5.4
Installing mime-types 2.99.1
Installing net-http-digest_auth 1.4
Installing net-http-persistent 2.9.4
Installing mini_portile2 2.0.0
Installing ntlm-http 0.1.1
Installing webrobots 0.1.2
Using bundler 1.12.5
Installing unf 0.1.4
Installing nokogiri 1.6.7.2
Installing domain_name 0.5.20160216
Installing http-cookie 1.0.2
Installing mechanize 2.7.4
Bundle complete! 3 Gemfile dependencies, 15 gems now installed.
Use `bundle show [gemname]` to see where a bundled gem is installed.
Post-install message from nokogiri:
Nokogiri is built with the packaged libraries: libxml2-2.9.2, libxslt-1.1.28, zlib-1.2.8, libiconv-1.14.

siberas@DESKTOP-J9Q9RTS MINGW64 ~/watobo/bin (master)
$ ruby watobo
#############################################################

     W A T O B O - THE Web Application Toolbox
     brought to you by siberas http://www.siberas.de

#############################################################
```

Up and running ;)

Please let me know if this works for you.

-andy

**Comment by siberas** [comment]
just released watobo gem version 0.9.23 ;)

**Comment by sfred92** [comment]
Hi Andy,

Ok, thank you for your great job.

I was planning to answer you, but you have been faster than me :)

Bye

Fred

De : siberas [mailto:notifications@github.com] 
Envoyé : vendredi 2 septembre 2016 15:54
À : siberas/watobo watobo@noreply.github.com
Cc : sfred92 fcrestin@free.fr; Author author@noreply.github.com
Objet : Re: [siberas/watobo] Crash while using proxy (#4)

just released watobo gem version 0.9.23 ;)
